extras/rya.export/export.accumulo/src/main/java/org/apache/rya/export/accumulo/AccumuloRyaStatementStore.java

```diff
@@ -0,0 +1,225 @@
+     * @param tablePrefix the Rya instance's table prefix.
+     * @param auths the comma-separated list of Accumulo authorizations for the
+     * user.
+     * @param zooKeepers the comma-separated list of zoo keeper host names.
+     * @throws MergerException
+     */
+    public AccumuloRyaStatementStore(final AccumuloRyaDAO dao, final String tablePrefix, final String ryaInstance) {
+        this.tablePrefix = tablePrefix;
+        if (tablePrefix != null) {
+            RdfCloudTripleStoreConstants.prefixTables(tablePrefix);
+        }
+        accumuloRyaDao = dao;
+        metadataRepo = new AccumuloParentMetadataRepository(dao);
+    }
+
+    @Override
+    public Iterator<RyaStatement> fetchStatements() throws FetchStatementException {
+        try {
+            final RyaTripleContext ryaTripleContext = RyaTripleContext.getInstance(accumuloRyaDao.getConf());
+
+            Scanner scanner = null;
+            try {
+                scanner = AccumuloRyaUtils.getScanner(tablePrefix + RdfCloudTripleStoreConstants.TBL_SPO_SUFFIX, accumuloRyaDao.getConf());
+                for (final IteratorSetting iteratorSetting : iteratorSettings) {
+                    scanner.addScanIterator(iteratorSetting);
+                }
+            } catch (final IOException e) {
+                throw new FetchStatementException("Unable to get scanner to fetch Rya Statements", e);
+            }
+            // Convert Entry iterator to RyaStatement iterator
+            final Iterator<Entry<Key, Value>> entryIter = scanner.iterator();
+            final Iterator<RyaStatement> ryaStatementIter = Iterators.transform(entryIter, new Function<Entry<Key, Value>, RyaStatement>() {
+               @Override
+               public RyaStatement apply(final Entry<Key, Value> entry) {
+                   final Key key = entry.getKey();
+                   final Value value = entry.getValue();
+                   RyaStatement ryaStatement = null;
+                   try {
+                       ryaStatement = AccumuloRyaUtils.createRyaStatement(key, value, ryaTripleContext);
+                   } catch (final TripleRowResolverException e) {
+                       log.error("Unable to convert the key/value pair into a Rya Statement", e);
+                   }
+                   return ryaStatement;
+               }
+            });
+            return ryaStatementIter;
+        } catch (final Exception e) {
+            throw new FetchStatementException("Failed to fetch statements.", e);
+        }
+    }
+
+    @Override
+    public void addStatement(final RyaStatement statement) throws AddStatementException {
+        try {
+            accumuloRyaDao.add(statement);
+            accumuloRyaDao.flush();
+            //This is a hack since a statement re-added with the same timestamp won't reappear since its been marked for deletion.
+            //RYA-197 is the ticket for fixing this hack.
+            if(!containsStatement(statement)) {
```

Comment:
this looks like you're querying before adding every statement.  That's really inefficient if so.